model/map/src/main/java/org/keycloak/models/map/common/Serialization.java

```diff
@@ -51,6 +52,7 @@
       .setSerializationInclusion(JsonInclude.Include.NON_NULL)
       .setVisibility(PropertyAccessor.ALL, Visibility.NONE)
       .setVisibility(PropertyAccessor.FIELD, Visibility.ANY)
+      .activateDefaultTyping(new LaissezFaireSubTypeValidator() /* TODO - see javadoc */, ObjectMapper.DefaultTyping.JAVA_LANG_OBJECT, JsonTypeInfo.As.PROPERTY)
```

Comment:
Why is this change necessary?

Reply:
To ensure that the JSONs contain proper typing of values in case of the `EnumMap<Field, Object>` which is necessary for deserialization. It adds type information into the serialized form of the `Object` values. When omitted, the values are deserialized into default collections, e.g. into a singleton read-only set rather than a proper `ArrayList`.

Reply:
I see. Thanks for the explanation.